app/models/service_ansible_playbook.rb

```diff
@@ -119,4 +120,11 @@ def inventory_name(action)
   def use_default_inventory?(hosts)
     hosts.blank? || hosts == 'localhost'
   end
+
+  # update job attributes only available to playbook provisioning
+  def update_job_for_playbook(action, job, hosts)
```

Comment:
This is a private method used internal only, but hosts might be `nil` if it is never set anywhere.